### PR TITLE
Fix schema.node_type_properties() and schema.rel_type_properties()

### LINKED ISSUE #1694: Implement int property in schema.node_type_properties()
In a graph where nodes of type :DifferentPropertyType have varying types for a property (string in one node and int in another), calling schema.node_type_properties() returns only _String_ as the property type, and not _[String, Int]_ as expected. 

[source](https://discord.com/channels/842007348272169002/890969285638369321/1204664686579027978)

### PR DESCRIPTION
**Pull request by DavIvek**
ATM mandatory field isn't working as expected. Also, if there are multiple types for one property, only one will be shown.
Will be fixed with this PR.

Logic behind the mandatory field is next:
Property is considered mandatory whenever all nodes or relationships of certain type have that property set to some value. Hence, if all relationships of certain type have certain property defined, than that property is considered mandatory and its value should be set to true (I'm checking that by comparing the number of occurrences of certain label and the number of occurrences of the property).

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

Release note: schema.node_type_properties() and schema.rel_type_properties() works as equivalents Neo4j's functions.

closes #1694 #1467

### CODE REVIEW

**Review by antoniofilipovic — COMMENTED** [review]
General comment, currently we are not spinning tests for schema.so, since you are working on this, do if you have time and it is right priority, you can check if everything is fine with schema e2e test, by brining it back here: https://github.com/memgraph/memgraph/pull/1675

File: query_modules/schema.cpp [hunk 1]
```diff
@@ -152,20 +144,23 @@ void Schema::NodeTypeProperties(mgp_list * /*args*/, mgp_graph *memgraph_graph,
       }
 
       if (node_types_properties.find(labels_set) == node_types_properties.end()) {
-        node_types_properties[labels_set] = PropertyInfo{std::set<Property, PropertyComparator>(), true};
+        node_types_properties[labels_set] = LabelsInfo();
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:147]: I think you don't need this line, as node_types_properties[labels_set] will construct if the entry doesn't exist: https://godbolt.org/z/9TzdEj1xK

Comment (antoniofilipovic) [reply query_modules/schema.cpp:147]: And link to cpp ref: https://en.cppreference.com/w/cpp/container/unordered_map/operator_at

File: query_modules/schema.cpp [hunk 2]
```diff
@@ -152,20 +144,23 @@ void Schema::NodeTypeProperties(mgp_list * /*args*/, mgp_graph *memgraph_graph,
...
       }
 
       auto &property_info = node_types_properties.at(labels_set);
-      for (auto &[key, prop] : node.Properties()) {
-        property_info.properties.emplace(key, std::move(prop));
-        if (property_info.mandatory) {
-          property_info.mandatory =
-              property_info.properties.size() == 1;  // if there is only one property, it is mandatory
+      for (auto const &[key, prop] : node.Properties()) {
+        const auto &prop_type = TypeOf(prop.Type());
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:158]: Be consistent here also, `auto const` then

File: query_modules/schema.cpp [hunk 3]
```diff
@@ -108,42 +109,33 @@ void Schema::ProcessPropertiesRel(mgp::Record &record, const std::string_view &t
   record.Insert(std::string(kReturnMandatory).c_str(), mandatory);
 }
 
-struct Property {
-  std::string name;
-  mgp::Value value;
+struct PropertyInfo {
+  std::unordered_set<std::string> values;
+  int64_t number_of_property_occurrences = 0;
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:114]: Is this number of property tuple occurances?

File: query_modules/schema.cpp [hunk 4]
```diff
@@ -197,40 +196,48 @@ void Schema::NodeTypeProperties(mgp_list * /*args*/, mgp_graph *memgraph_graph,
...
+          property_info.properties[key].values.emplace(prop_type);
+          property_info.properties[key].number_of_property_occurrences++;
         }
       }
     }
 
     for (auto &[type, property_info] : rel_types_properties) {
       std::string type_str = ":`" + std::string(type) + "`";
       for (auto const &prop : property_info.properties) {
+        mgp::List values = mgp::List();
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:230]: `auto values = mgp::List()`, use `auto` not to duplicate type

File: query_modules/schema.cpp [hunk 5]
```diff
@@ -152,20 +144,23 @@ void Schema::NodeTypeProperties(mgp_list * /*args*/, mgp_graph *memgraph_graph,
...
       }
 
+      node_types_properties[labels_set].number_of_label_occurrences++;
+
       if (node.Properties().empty()) {
-        node_types_properties[labels_set].mandatory = false;  // if there is node with no property, it is not mandatory
         continue;
       }
 
       auto &property_info = node_types_properties.at(labels_set);
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:156]: Isn't this labels_info?

File: query_modules/schema.cpp [hunk 6]
```diff
@@ -108,42 +109,33 @@ void Schema::ProcessPropertiesRel(mgp::Record &record, const std::string_view &t
   record.Insert(std::string(kReturnMandatory).c_str(), mandatory);
 }
 
-struct Property {
-  std::string name;
-  mgp::Value value;
+struct PropertyInfo {
+  std::unordered_set<std::string> values;
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:113]: These are property types?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few comments, did you document already on docs how this algorithm exactly works?

File: query_modules/schema.cpp (hunk 3 above)
Comment (DavIvek) [reply query_modules/schema.cpp:114]: This counter is used to check how many times a property has appeared with a certain label. If the number of occurrences of given property is the same as number of occurrences for a certain label then all labels with that name have this property, hence the property is mandatory.

File: query_modules/schema.cpp (hunk 6 above)
Comment (DavIvek) [reply query_modules/schema.cpp:113]: Yes, I'll rename it to be clearer.

**Comment by DavIvek** [comment]
I didn't document it already, I will when this get merged.

**Comment by katarinasupe** [comment]
@DavIvek is this going to Memgraph 2.15? If yes, can you add milestone?

**Comment by DavIvek** [comment]
@katarinasupe I hope, if @antoniofilipovic manages to leave a review tomorrow, I know that he has a lot of things going on HA.

**Comment by kgolubic** [comment]
@DavIvek please don't forget about the docs and release notes.

**Review by antoniofilipovic — APPROVED** [review]

File: query_modules/schema.cpp [hunk 7]
```diff
@@ -108,83 +109,78 @@ void Schema::ProcessPropertiesRel(mgp::Record &record, const std::string_view &t
...
-struct Property {
-  std::string name;
-  mgp::Value value;
+struct PropertyInfo {
+  std::unordered_set<std::string> property_types;  // property types
+  int64_t number_of_property_occurrences = 0;
+
+  PropertyInfo() = default;
+  explicit PropertyInfo(std::string property_type)
+      : property_types({property_type}), number_of_property_occurrences(1) {}
```
Comment (antoniofilipovic) [inline query_modules/schema.cpp:118]: Not a mandatory change, but I would do following
```suggestion
      : property_types({std::move(property_type)}), number_of_property_occurrences(kInitialNumberOfPropertyOccurances) {}
```

**Comment by katarinasupe** [comment]
> This PR will be a part of the next release.

@DavIvek then issue https://github.com/memgraph/memgraph/issues/1467 isn't a part of the milestone 2.15 anymore? Can you update that?

**Comment by kgolubic** [comment]
@DavIvek I can see the RN, but I can't find the docs. Is there a docs PR?

**Pull request merged**